lib/b3fs/src/hasher/byte_hasher/mod.rs

```diff
@@ -44,6 +44,9 @@ pub struct Blake3Hasher<T: HashTreeCollector = Vec<[u8; 32]>> {
     pub tree: T,
 }
 
+unsafe impl<T: HashTreeCollector> Send for Blake3Hasher<T> {}
+unsafe impl<T: HashTreeCollector> Sync for Blake3Hasher<T> {}
```

Comment:
this is not sound if `T: !Send`

```suggestion
unsafe impl<T: HashTreeCollector> Send for Blake3Hasher<T> where T: Send {}
unsafe impl<T: HashTreeCollector> Sync for Blake3Hasher<T> where T: Sync {}
```